samples/core/Multitenancy/exampleschema/Program.cs

```diff
@@ -0,0 +1,182 @@
+    public class Contact
+    {
+        public Guid Id { get; set; } = Guid.NewGuid();
+        public string Name { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantAContact : Contact { }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantBContact : Contact { }
+
+    /// <summary>
+    /// Using IQueryable.
+    /// </summary>
+    public class ContactContext : DbContext
+    {
+        private readonly string _tenant;
+
+        public ContactContext(
+            TenantProvider provider,
+            DbContextOptions<ContactContext> opts)
+            : base(opts) => _tenant = provider.Tenant;
+
+        public DbSet<TenantAContact> TenantAContacts { get; set; } = null!;
+        public DbSet<TenantBContact> TenantBContacts { get; set; } = null!;
+
+        public IQueryable<Contact> Contacts => _tenant == "TenantA" ?
+            TenantAContacts : TenantBContacts;
+
+        // SQLite doesn't handle multiple schemas, but for SQLServer you
+        // can uncomment the code below for TenantaA.Contact and TenantB.Contact
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            //modelBuilder.Entity<TenantAContact>()
+            //    .ToTable(nameof(Contact), "TenantA");
+
+            //modelBuilder.Entity<TenantBContact>()
+            //    .ToTable(nameof(Contact), "TenantB");
+
+            base.OnModelCreating(modelBuilder);
+        }
+    }
+
+    /// <summary>
+    /// Using multiple models with schema/name overrides.
+    /// </summary>
+    public class AlternateContactContext : DbContext
+    {
+        public string Tenant { get; set; } = null!;
+
+        public AlternateContactContext(
+            TenantProvider provider,
+            DbContextOptions<AlternateContactContext> opts)
+            : base(opts) => Tenant = provider.Tenant;
+
+        public DbSet<Contact> Contacts { get; set; } = null!;
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+         =>
+            // SQLite doesn't handle multiple schemas, so we use different
+            // table names but the same API can be used with SQLServer for
+            // same table name, different schema
+            modelBuilder.Entity<Contact>().ToTable($"{Tenant}Contacts");
+
+        /// <summary>
+        /// Tell EF Core to cache the model by tenant.
+        /// </summary>
+        /// <param name="optionsBuilder">The options builder.</param>
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder
+            .ReplaceService<IModelCacheKeyFactory, TenantModelCacheKeyFactory>();
+    }
+
+    /// <summary>
+    /// Implemented to support a model per tenant in the same DbContext.
+    /// </summary>
+    public class TenantModelCacheKeyFactory : IModelCacheKeyFactory
```

Comment:
See the new EF Core 6.0 note in the [docs](https://docs.microsoft.com/en-us/ef/core/modeling/dynamic-model#imodelcachekeyfactory) on the design-time bit.